Updates Zone09 scenarios to match updated specification (#1517)
dotse/zonemaster: docs/public/specifications/test-zones/Zone-TP/zone09.md

> # Specification of Test Scenariso for ZONE09

Reviewer: ```suggestion
# Specification of Test Scenarios for ZONE09
```

* Z09_MX_FOUND
* Z09_NON_AUTH_MX_RESPONSE
* Z09_NO_MX_FOUND
* Z09_NO_MX_FOUND_OR_EXPECTED
* Z09_NO_SERVERS_MX_RESPONSE
* Z09_NO_RESPONSE_MX_QUERY
* Z09_NULL_MX_NON_ZERO_PREF
* Z09_NULL_MX_WITH_OTHER_MX
* Z09_ROOT_EMAIL_DOMAIN
* Z09_TLD_EMAIL_DOMAIN
* Z09_UNEXPECTED_RCODE_MX
* Z09_VALID_NULL_MX

## Test scenarios and message tags

If a message tag is not listed for the scenario, its presence or non-presence is
irrelevant to the test scenario and must be ignored.

| Scenario name          | Mandatory message tags                                              | Forbidden message tags |
|:-----------------------|:--------------------------------------------------------------------|:-----------------------|
| NO-RESPONSE-MX-QUERY-1 | Z09_NO_RESPONSE_MX_QUERY, Z09_MX_DATA                               | 2)                     |
| NO-RESPONSE-MX-QUERY-2 | Z09_NO_RESPONSE_MX_QUERY, Z09_NO_SERVERS_MX_RESPONSE                | 2)                     |
| UNEXPECTED-RCODE-MX    | Z09_UNEXPECTED_RCODE_MX, Z09_MISSING_MAIL_EXCHANGE                  | 2)                     |
| NON-AUTH-MX-RESPONSE   | Z09_NON_AUTH_MX_RESPONSE, Z09_MX_DATA                               | 2)                     |

> | INCONSISTENT-MX        | Z09_INCONSISTENT_MX, Z09_MX_FOUND Z09, Z09_NO_MX_FOUND, Z09_MX_DATA | 2)                     |

Reviewer: ```suggestion
| INCONSISTENT-MX        | Z09_INCONSISTENT_MX, Z09_MX_FOUND, Z09_NO_MX_FOUND, Z09_MX_DATA    | 2)                     |
```

| INCONSISTENT-MX-DATA-1 | Z09_INCONSISTENT_MX_DATA, Z09_MX_DATA                               | 2)                     |
| INCONSISTENT-MX-DATA-1 | Z09_INCONSISTENT_MX_DATA, Z09_MX_DATA                               | 2)                     |

> | INCONSISTENT-MX-DATA-1 | Z09_INCONSISTENT_MX_DATA, Z09_MX_DATA                               | 2)                     |

Reviewer: ```suggestion
| INCONSISTENT-MX-DATA-1 | Z09_INCONSISTENT_MX_DATA, Z09_MX_DATA                               | 2)                     |
| INCONSISTENT-MX-DATA-2 | Z09_INCONSISTENT_MX_DATA, Z09_MX_DATA                               | 2)                     |
| INCONSISTENT-MX-DATA-3 | Z09_INCONSISTENT_MX_DATA, Z09_MX_DATA                               | 2)                     |
```

  without) MX records on MX queries.
* Unless otherwise stated, all responses are authoritative and with [RCODE Name]
  "NoError".
* Unless otherwise stated all name servers for the zone responds with an
  identical, non-empty, non-Null MX RRset on MX query.
* Unless otherwise stated all zones are served by two name severs whose name are
  in-domain.

### NO-RESPONSE-MX-QUERY-1
* Zone: "no-response-mx-query-1.zone09.xa."
  * One name server does not respond on MX query.

### NO-RESPONSE-MX-QUERY-2
* Zone: "no-response-mx-query-2.zone09.xa."
  * No name server responds on MX query.

### UNEXPECTED-RCODE-MX
* Zone: "unexpected-rcode-mx.zone09.xa."
  * One name server returns with any [RCODE Name] except "NoError".
  * The other returns no MX.

### NON-AUTH-MX-RESPONSE
* Zone: "non-auth-mx-response.zone09.xa."
  * One name server returns with [RCODE Name] "NoError" and non-AA on MX query.

### INCONSISTENT-MX
* Zone: "inconsistent-mx.zone09.xa."
  * One name server responds without MX RRset (NODATA).

### INCONSISTENT-MX-DATA-1
* Zone: "inconsistent-mx-data-1.zone09.xa."

>   * The name servers repond with MX RRsets that are not equal.

Reviewer: ```suggestion
  * The name servers respond with MX RRsets that are not equal.
```

The same typo appears in other places.

### INCONSISTENT-MX-DATA-2
* Zone: "inconsistent-mx-data-2.zone09.xa."
  * The name servers repond with the same mail exchange names, but with
    different preferences.

### INCONSISTENT-MX-DATA-3
* Zone: "inconsistent-mx-data-3.zone09.xa."
  * The name servers repond with the same mail exchange names, but with

>     different preferences.

Reviewer: What’s the difference between this scenario and the previous one, then?

Author: Updated.

### MIXED-TTL-1
* Zone: "mixed-ttl-1.zone09.xa."
  * The TTL of the MX RRset are different on the two NS.

### MIXED-TTL-2
* Zone: "mixed-ttl-1.zone09.xa."

>   * The TTL of the records in the MX RRset are different.

Reviewer: ```suggestion
* Zone: "mixed-ttl-2.zone09.xa."
  * The TTL of the records in the MX RRset are different.
```

  without) MX records on MX queries.
* Unless otherwise stated, all responses are authoritative and with [RCODE Name]
  "NoError".

> * Unless otherwise stated all name servers for the zone responds with an

Reviewer: ```suggestion
* Unless otherwise stated all name servers for the zone respond with an
```

  identical, non-empty, non-Null MX RRset on MX query.

> * Unless otherwise stated all zones are served by two name severs whose name are

Reviewer: ```suggestion
* Unless otherwise stated all zones are served by two name servers whose name are
```

  in-domain.

### NO-RESPONSE-MX-QUERY-1
* Zone: "no-response-mx-query-1.zone09.xa."
  * One name server does not respond on MX query.

### NO-RESPONSE-MX-QUERY-2
* Zone: "no-response-mx-query-2.zone09.xa."
  * No name server responds on MX query.

### UNEXPECTED-RCODE-MX
* Zone: "unexpected-rcode-mx.zone09.xa."
  * One name server returns with any [RCODE Name] except "NoError".
  * The other returns no MX.

### NON-AUTH-MX-RESPONSE
* Zone: "non-auth-mx-response.zone09.xa."
  * One name server returns with [RCODE Name] "NoError" and non-AA on MX query.

### INCONSISTENT-MX
* Zone: "inconsistent-mx.zone09.xa."
  * One name server responds without MX RRset (NODATA).

### INCONSISTENT-MX-DATA-1
* Zone: "inconsistent-mx-data-1.zone09.xa."
  * The name servers respond with MX RRsets that are not equal, two MX records

>     from one server, and just one of the two MX records from the other.

Reviewer: Just to make it a little bit clearer:
```suggestion
  * The name servers respond with MX RRsets that are not equal: one server gives
    two MX records, while the other one gives just one of the two MX records.
```

Author: Is colon really correct? I think it should be semi-colon or a period. I'll take your update, but change to period.

### INCONSISTENT-MX-DATA-2
* Zone: "inconsistent-mx-data-2.zone09.xa."
  * Both name servers respond with one MX record which has the same mail exchange

>     names from both servers, but the preferences different.

Reviewer: If I understand correctly…?
```suggestion
  * Both name servers respond with MX RRsets that are not equal: one server gives
    one MX record, while the other one gives an MX record with the same mail
    exchange name but a different preference value.
```

Author: Same here.

### INCONSISTENT-MX-DATA-3
* Zone: "inconsistent-mx-data-3.zone09.xa."
  * Both name servers respond with two MX records and the same two mail exchange
    names are used, but the orders between the the two records, as defined by

>     the preference, are different between the two servers.

Reviewer: I see. How about:
```suggestion
  * Both name servers respond with MX RRsets that are not equal: one server
    gives two MX records, while the other one gives two MX records that list the
    same mail exchange names but with different preference values, such that
    when sorting by preference value, the mail exchanges are in a different order
    than the first server’s MX RRset.
```

Author: Same here.

Author: I think this description is too long. The description is for some one to implement, or check, the scenarios. I'll cut the last part off.
